lib/parse.js

```diff
@@ -12,20 +15,33 @@ exports = module.exports = function(content, options) {
       root = exports.evaluate('<root></root>', options)[0];
 
   root.type = 'root';
+  root.parent = null;
 
   // Update the dom using the root
   exports.update(dom, root);
 
   return root;
 };
 
+function parseWithParse5 (content) {
+  var parseAsDocument = /^(\s|<!--.*?-->)*?<(!doctype|html|head|body)(.*?)>/i.test(content),
```

Comment:
Speaking clearly it just enabled the behavior that we currently have in cheerio. Unless we'll not add explicit API for document and fragment parsing, but as far as I can tell it will lead to breaking change.